pyhealth/datasets/base_dataset.py

```diff
@@ -386,17 +422,22 @@ def get_all_tokens(
             raise ValueError("Please set task first.")
         tokens = []
         for sample in self.samples:
+            if type(sample[key]) == list:
+                if len(sample[key]) == 0:
+                    continue
+                # a list of lists of values
+                elif type(sample[key][0]) == list:
+                    tokens.extend(sum(sample[key], []))
+                # a list of values
+                else:
+                    tokens.extend(sample[key])
             # single value
-            if type(sample[key]) in [bool, int, str]:
-                tokens.append(sample[key])
-            # a list of values
-            elif type(sample[key][0]) in [bool, int, str]:
-                tokens.extend(sample[key])
-            # a list of lists of values
-            elif type(sample[key][0]) == list:
-                tokens.extend(sum(sample[key], []))
             else:
```

Comment:
Here, we only consider the list and list of list cases. However, maybe the `is_homo_list` function does not need that recursion.

Reply:
Do you mean `list_nested_level()`? I try to make `list_nested_level()` more general instead of only applying to two level nested list.

Reply:
ok